src/components/PracticalExperience.js

```diff
@@ -1,63 +1,136 @@
 import React, { useState } from 'react';
 
 const PracticalExperience = ({ onSubmit }) => {
-  const [editMode, setEditMode] = useState(true);
-  const [experience, setExperience] = useState({
+  const [jobs, setJobs] = useState([]);
+  const [newJob, setNewJob] = useState({
     companyName: '',
     positionTitle: '',
     mainResponsibilities: '',
     dateFrom: '',
     dateUntil: '',
   });
+  const [editMode, setEditMode] = useState(false);
 
   const handleChange = (e) => {
-    setExperience({ ...experience, [e.target.name]: e.target.value });
+    setNewJob({ ...newJob, [e.target.name]: e.target.value });
   };
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    setEditMode(false);
-    onSubmit(experience);
+
+    // Date validation
+    if (!isValidDateRange(newJob.dateFrom, newJob.dateUntil)) {
+      alert('Date Until cannot be before Date From');
+      return;
+    }
+
+    // Add new job to jobs array
+    setJobs([...jobs, newJob]);
+
+    // Clear the form for the next entry
+    setNewJob({
+      companyName: '',
+      positionTitle: '',
+      mainResponsibilities: '',
+      dateFrom: '',
+      dateUntil: '',
+    });
   };
 
+  const isValidDateRange = (dateFrom, dateUntil) => {
+    return new Date(dateUntil) >= new Date(dateFrom);
+  };
+
+  const handleEdit = (index) => {
+    const jobToEdit = jobs[index];
+    setNewJob({
+      companyName: jobToEdit.companyName,
+      positionTitle: jobToEdit.positionTitle,
+      mainResponsibilities: jobToEdit.mainResponsibilities,
+      dateFrom: jobToEdit.dateFrom,
+      dateUntil: jobToEdit.dateUntil,
+    });
+    // Remove the job from the list to be edited
+    setJobs(jobs.filter((job, idx) => idx !== index));
+
```

Comment:
If I start to edit a job & then press cancel, this code will cause the job to dissapear right?